engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/CreateDeploymentTest.java

```diff
@@ -536,6 +538,116 @@ public void shouldCreateDeploymentResourceOfExecutablePools() {
         .doesNotContain("Process_NonExecutable");
   }
 
+  @Test
+  public void shouldCreateProcessForTenant() {
+    // given
+    final String tenant = "tenant";
+
+    // when
+    final var deployment =
+        ENGINE.deployment().withXmlResource(process).withTenantId(tenant).deploy();
+
+    // then
+    assertThat(deployment.getValue().getTenantId()).isEqualTo(tenant);
+    assertThat(
+            RecordingExporter.processRecords()
+                .withIntent(ProcessIntent.CREATED)
+                .withBpmnProcessId(processId)
+                .limit(1))
+        .extracting(Record::getValue)
+        .extracting(
+            ProcessMetadataValue::getBpmnProcessId,
+            ProcessMetadataValue::getVersion,
+            ProcessMetadataValue::getProcessDefinitionKey,
+            TenantOwned::getTenantId)
+        .describedAs("Process is created for correct tenant")
+        .containsExactly(
+            tuple(
+                processId,
+                1,
+                deployment.getValue().getProcessesMetadata().get(0).getProcessDefinitionKey(),
+                tenant));
+  }
+
+  @Test
+  public void shouldCreateProcessesForTenant() {
+    // given
+    final String tenant = "tenant";
+
+    // when
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(process)
+            .withXmlResource(process2)
+            .withTenantId(tenant)
+            .deploy();
+
+    // then
+    assertThat(deployment.getValue().getTenantId()).isEqualTo(tenant);
+    assertThat(RecordingExporter.processRecords().withIntent(ProcessIntent.CREATED).limit(2))
+        .extracting(Record::getValue)
+        .extracting(
+            ProcessMetadataValue::getBpmnProcessId,
+            ProcessMetadataValue::getVersion,
+            ProcessMetadataValue::getProcessDefinitionKey,
+            TenantOwned::getTenantId)
+        .describedAs("Processes are created for correct tenant")
+        .containsExactly(
+            tuple(
+                processId,
+                1,
+                deployment.getValue().getProcessesMetadata().get(0).getProcessDefinitionKey(),
+                tenant),
+            tuple(
+                processId2,
+                1,
+                deployment.getValue().getProcessesMetadata().get(1).getProcessDefinitionKey(),
+                tenant));
+  }
+
+  @Test
+  @Ignore(
+      "Unignore in 13320 part 2. Currently 2nd deployment is marked as duplicate as the state is not supporting tenants yet.")
+  public void shouldCreateProcessesForTenants() {
+    // given
+    final String tenant = "tenant";
+    final String tenant2 = "tenant2";
+
+    // when
+    final var deployment =
+        ENGINE.deployment().withXmlResource(process).withTenantId(tenant).deploy();
+    final var deployment2 =
+        ENGINE.deployment().withXmlResource(process).withTenantId(tenant2).deploy();
```

Comment:
🔧 Let's show that these are considered non-duplicate by asserting that the deployment key is different and the process definition keys differ